### ISSUE TITLE
Build Environment Detective Agent with Docker generation

### ISSUE THREAD

**Issue by armaanamatya** [armaanamatya/openresearch#24]
## Overview

Build the Environment Detective Agent — the agent that infers Python/CUDA/framework versions, recovers old package versions, and generates a working Dockerfile. This is one of the hardest technical problems per the PRD.

**PRD references:** Environment Detective Agent, Demo Papers (pre-solved environments for PPO and MixMatch).

### 1. Agent prompt
System prompt for `environment-detective` that:
- Reads workspace variables: `paper_text`, `repo_files`, `configs`, `requirements`, `github_issues`
- Uses `Read`, `Write`, `Bash`, `WebSearch` tools
- Produces environment spec with Dockerfile

### 2. Environment inference logic
The agent should:
- Grep paper for version mentions (CUDA, PyTorch, Python, framework)
- Check requirements.txt / setup.py / conda env files
- Cross-reference GitHub issues for installation clues
- Infer compatible version matrix
- Handle simulator versions for robotics papers (MuJoCo, Gymnasium, etc.)

### 3. Dockerfile generation
Output a working Dockerfile:
- Base image selection
- System packages
- Python environment + pip installs with pinned versions
- Framework installation (PyTorch, TensorFlow, JAX)
- NVIDIA Container Toolkit flags for GPU (`--gpus`, `--shm-size=8g`)

### 4. Environment assumption tracking
- Record each inferred version as an assumption with risk level
- Submit assumptions to orchestrator's append API
- Include evidence citations for each assumption

### 5. PPO demo validation
- Given PPO paper context, agent should produce a Dockerfile matching:
  ```dockerfile
  FROM python:3.11-slim
  RUN pip install torch==2.2.0 --index-url https://download.pytorch.org/whl/cpu
  RUN pip install gymnasium==0.29.1 numpy==1.26.4
  ```

## Acceptance Criteria
- [ ] Agent produces working Dockerfile for PPO demo paper
- [ ] Version inference uses multiple evidence sources (paper, requirements, issues)
- [ ] Each version choice has an assumption ledger entry with evidence
- [ ] Dockerfile builds successfully (`docker build` passes)
- [ ] Environment spec written to workspace as `environment_spec` variable with citations

## Dependencies
- #22 (orchestrator)
- #23 (paper understanding agent provides claim_map context)

## Labels
`priority:p0`, `area:backend`, `area:agents`, `type:task`
